### PR TITLE
Port to dot net standard 2.0

### LINKED ISSUE #219: Make XAML Styler console app cross-platform
See details in #217.

### PR DESCRIPTION
**Pull request by zachdean**
Hi Everyone. Below is a PR to port this project to dotnet Standard.

**What**: This fork is a port from dotnet framework to dotnet standard 2.0 in the XamlStyler.Core. 

**Why**: To allow for XamlStyler.Console to be cross-platform.

**Comments**:
XamlStyler.Console was ported to a dotnet core 3.0 application to allow XamlStyler.Console to be used on MacOS. We are currently testing the port on my team (we are a mix of Windows and Mac users) as a pre-commit hook and so far the console and the VS 2019 plugin have been stable with limited testing. This port is not complete, but more of a proof of concept to see if this is the direction we want to take the project.

**Tested**:
XamlStyler.Console - MacOS and Winx64
XamlStyler.Package (note: this package has not changed) - Visual Studio 2019

**TODO's**:
1. change XamlStyler.Console Nuget package build script to target winx32, winx64, macOs and put in package together
2. regression testing on VS 2019, 2017 and VS for Mac
3. test build pipelines to make sure no tooling was broken

Fixes #219

### CODE REVIEW

**Comment by grochocki** [comment]
It certainly would be great for XamlStyler.Console to work cross-platform. Thanks for taking up this initiative! Assuming testing goes well and existing scenarios do not regress, this seems like a good direction.

**Comment by zachdean** [comment]
I have updated the nuspec and build nuget build bat to create the different run-times and package it up. The nuget package is published to \build\nuget\nupkgs. We have been running the port internally as a pre-commit hook on both Windows and Mac with no problems. I have also been running the extension on VS 2019 with no issues. Note: to build console you need to have dotnet core 3.0 sdk installed on your machine.

**Comment by grochocki** [comment]
Thanks, @zachdean! I am not able to run unit tests inside of Visual Studio. It also looks like unit tests aren't running in [pipelines](https://dev.azure.com/xavalon/XAML%20Styler/_build/results?buildId=107&view=ms.vss-test-web.build-test-results-tab) either. Am I missing a step here?

File: build/nuget/XamlStyler.Console.nuspec [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <package xmlns="http://schemas.microsoft.com/packaging/2011/10/nuspec.xsd">
 	<metadata>
 		<id>XamlStyler.Console</id>
-		<version>3.0.0.0</version>
+		<version>4.0.0.0</version>
```
Comment (grochocki) [inline build/nuget/XamlStyler.Console.nuspec]: While this is certainly a major change for the console app, we'd like to keep versions in lockstep with the extension to make it easier for compatibility checks and debugging. Can we snap to 3.3.0.0? I'd like to eventually align these via build scripts in pipelines.

**Comment by zachdean** [comment]
@grochocki I ran into a issue with Visual Studio this week were it did not run unit tests unless dotnet core 2.2 runtime was installed (one box wanted x86, the other wanted x64, not sure how it was picking the architecture since they both were set to run as x86). I think it is a vs bug since it should be using dotnet 3.0. However, the tests run once the runtime or sdk is installed.

**Comment by zachdean** [comment]
@grochocki I updated the build pipeline and rebased to fix the merge conflicts. You should see the tests in the pipeline now. After this is merged into master I will make some more pipelines to automate the build of nuget package and extension so that we just need to create a release pipeline that consumes the build.

**Comment by zachdean** [comment]
@grochocki I normalized the build pipeline a little bit to make it easier to see what is failing. I also added the build pipeline for the nuget package so that it can be used as dotnet tool (#218, I figured it would be better to start this way instead of the self contained exe for each platform that is was before). The pipeline will auto increment the nuget package in preparation for adding a continuous delivery pipeline for the package.

File: XamlStyler.Console/XamlStyler.Console.csproj [hunk 2]
```diff
@@ -1,109 +1,40 @@
...
-    <ApplicationVersion>1.0.0.%2a</ApplicationVersion>
-    <IsWebBootstrapper>false</IsWebBootstrapper>
-    <UseApplicationTrust>false</UseApplicationTrust>
-    <BootstrapperEnabled>true</BootstrapperEnabled>
+    <PackageId>XamlStyler.Console</PackageId>
+    <Version>3.3.0.0</Version>
+    <Authors>Xavalon</Authors>
+    <Company>Xavalon</Company>
+    <Description>XAML Styler is a visual studio extension, which formats XAML source code by sorting the attributes based on their importance. This tool can help you/your team maintain a better XAML coding style as well as a much better XAML readability.</Description>
+    <Copyright>Copyright © Xavalon 2016</Copyright>
```
Comment (grochocki) [inline XamlStyler.Console/XamlStyler.Console.csproj]: While we are in here, can you update this to 2019?

File: XamlStyler.Core/XamlStyler.Core.csproj [hunk 3]
```diff
@@ -1,134 +1,30 @@
...
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-  </PropertyGroup>
-  <PropertyGroup>
     <SignAssembly>true</SignAssembly>
-  </PropertyGroup>
-  <PropertyGroup>
     <AssemblyOriginatorKeyFile>Key.snk</AssemblyOriginatorKeyFile>
+    <DelaySign>false</DelaySign>
+    <Copyright>Copyright ©  2012</Copyright>
```
Comment (grochocki) [inline XamlStyler.Core/XamlStyler.Core.csproj]: "Copyright © Xavalon 2019"

File: XamlStyler.Console/XamlStyler.Console.csproj [hunk 4]
```diff
@@ -1,109 +1,40 @@
...
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProjectGuid>{5284FEA7-FC0B-45EA-8B16-9DA7F5E6D7B6}</ProjectGuid>
     <OutputType>Exe</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <TargetFramework>netcoreapp2.2</TargetFramework>
```
Comment (grochocki) [inline XamlStyler.Console/XamlStyler.Console.csproj]: Can we target 3.1? I tried this out on my machine, and it seems to eliminate the need to install 2.2 run time. 3.1 is also an LTS version and contains some useful features for packaging + reducing binary size. https://devblogs.microsoft.com/dotnet/announcing-net-core-3-1/

File: XamlStyler.UnitTests/XamlStyler.UnitTest.csproj [hunk 5]
```diff
@@ -1,369 +1,19 @@
...
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProductVersion>
-    </ProductVersion>
-    <SchemaVersion>2.0</SchemaVersion>
-    <ProjectGuid>{E5245034-9CD0-4FC6-B141-BAE17BC251A5}</ProjectGuid>
-    <OutputType>Library</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <TargetFramework>netcoreapp2.2</TargetFramework>
```
Comment (grochocki) [inline XamlStyler.UnitTests/XamlStyler.UnitTest.csproj]: 3.1 here too

File: build/pipelines/templates/build-extension-public.yaml [hunk 6]
```diff
@@ -2,16 +2,23 @@
...
     clean: true
-  
+
+  - task: PowerShell@2
+    displayName: Update version in the vsix manifest
+    inputs:
+      filePath: 'build\update-version.ps1'
+      arguments: $(version)
+      pwsh: true
+
```
Comment (zachdean) [inline build/pipelines/templates/build-extension-public.yaml]: run powershell script to keep the extension version in sync with the console

File: build/pipelines/templates/build-console-public.yaml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+      command: 'pack'
+      packagesToPack: '**/XamlStyler.Console.csproj'
+      arguments: '--configuration $(buildConfiguration)'
+      versioningScheme: 'off'
+      nobuild: true
+      majorVersion: '$(major)'
+      minorVersion: '$(minor)'
+      patchVersion: '$(patch)'
+
+  - task: PublishPipelineArtifact@1
```
Comment (grochocki) [inline build/pipelines/templates/build-console-public.yaml]: I am not finding this artifact published anywhere. I do see the VSIX (with the right version number), which seems to install and work properly.

File: build/pipelines/templates/build-console-public.yaml [hunk 8]
```diff
@@ -39,5 +39,4 @@ jobs:
     displayName: 'Publish XamlStyler.Console artifact'
     inputs:
       PathtoPublish: '$(Build.ArtifactStagingDirectory)'
-      ArtifactName: 'XamlStyler.Console'
-      publishLocation: 'Container'
\ No newline at end of file
+      ArtifactName: 'XamlStyler.Console'
```
Comment (grochocki) [inline build/pipelines/templates/build-console-public.yaml]: Hmm, I am seeing a new artifact now, but it appears to be the entire source tree + some additional folders. I _think_ we just want the nupkg output from 'a' sub-directory, right?

Comment (zachdean) [reply build/pipelines/templates/build-console-public.yaml]: just pushed a commit to fix that

**Comment by zachdean** [comment]
@grochocki The build will output only the vsix and the nuget package now instead of everthing.

**Pull request merged**